crates/walrus-service/src/event/event_blob_downloader.rs

```diff
@@ -259,4 +220,112 @@ impl EventBlobDownloader {
 
         Ok(blobs)
     }
+
+    /// Reads the status of an event blob.
+    ///
+    /// When reading the first certified event blob, it may be the case that the event blob is
+    /// just certified, and the storage nodes may not know it yet. Therefore, upon encountering
+    /// a nonexistent blob, we will retry and wait for the blob to be certified.
+    /// For all the earlier blobs, they must have been seen by the storage nodes, since
+    /// otherwise, no new certified event blobs would be created.
+    async fn read_event_blob_status(
+        &self,
+        event_blob_id: BlobId,
+        reading_first_event_blob: bool,
+        start_time: Instant,
+    ) -> Result<BlobStatus> {
+        let blob_status = loop {
```

Comment:
Remark: Like @sadhansood, I find the double retries for (1) *getting the status* and (2) *reading* the first event blob a bit unsatisfying.

Besides skipping the status check as Sadhan suggested, there would be another option: We could remove or add a parameter to skip the [`verify_blob_status_event` in the `get_verified_blob_status` function](https://github.com/MystenLabs/walrus/blob/22481330ca235faef6351869e65e3d499c4c25fe/crates/walrus-sdk/src/client.rs#L2140). If we do that and get a certified status, we *know* that a quorum of nodes reported that status. Then we could remove the retries for reading the blob.